### PR TITLE
NGSTACK-315 add basic deployer configuration

### PR DESCRIPTION
**Pull request by iherak**
Files that need to be changed per project are:
* `deploy/hosts.php`
* `deploy/parameters.php`

For server specific configuration, edit the `deploy/hosts.php` file and add proper hostname, adjust users etc.
For project specific configuration, edit the `deploy/parameters.php` file with project specifics (app name, repository, branch, sentry configuration etc)

### CODE REVIEW

File: deploy/tasks/deploy_log.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    $commitData = run('cd {{release_path}} && git log origin/{{branch}} -1');
+    $string = getDeployLogInformation($commitData);
+
+    $tmpFile = tempnam(sys_get_temp_dir(), 'tmp_deploy');
+    file_put_contents($tmpFile, $string);
+    upload($tmpFile, $newReleaseFilename);
+    unlink($tmpFile);
+
+    $oldestRelease = $releasesList[get('keep_releases')-1] - 1;
```
Comment (iherak) [inline deploy/tasks/deploy_log.php]: This fails on initial deploys. Thank you @RandyCupic for reporting.

File: deploy/tasks/server.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+desc('Symlink web folder to appropriate place');
+task('server:symlink_web', function() {
+    run("{{bin/symlink}} {{release_path}}/web {{deploy_path}}");
+});
+
+set('varnish_ban_hosts', []);
+desc('Invalidate content on varnish per host');
+option('ban_host', null, InputOption::VALUE_REQUIRED, 'Overrides varnish_ban_hosts configuration and uses provided value');
+task('varnish:ban', function() {
```
Comment (iherak) [inline deploy/tasks/server.php]: todo: add option whether to use varnishadm or curl (in case varnish is on a different server)

File: deploy.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    'server:symlink_web',
+    'cachetool:clear:opcache',
+    'deploy:unlock',
+    'cleanup',
+])->desc('Deploy your project');
+
+// after successful deploy
+after('deploy', 'varnish:ban');
+after('deploy', 'deploy:log:remote');
+
```
Comment (RandyCupic) [inline deploy.php]: @iherak I think that there's missing the following: `after('deploy', 'success');`. There are some tasks (eg. Slack notify success) which are being triggered after `success` but apparently this one has to be executed manually at the end of deploy procedure. After adding that, Slack success notify is working correctly and I'm also getting a `Successfully deployed!` message after the deploy is finished.

File: deploy/tasks/deploy_log.php (hunk 1 above)
Comment (iherak) [reply deploy/tasks/deploy_log.php]: Should be fixed by 359f3757946c0fcde0c2845573cb6f33e20141c5

File: deploy/tasks/server.php (hunk 2 above)
Comment (iherak) [reply deploy/tasks/server.php]: Added option to use fos http cache and invalidate tags instead.

File: deploy.php (hunk 3 above)
Comment (iherak) [reply deploy.php]: Done in 359f3757946c0fcde0c2845573cb6f33e20141c5

**Comment by iherak** [comment]
@emodric @RandyCupic this should now be ready for a round of reviews.

File: deploy/tasks/server.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        invalidateVarnish();
+    }
+});
+
+
+set('apache_use_sudo', true);
+desc('Reload Apache');
+task('server:reload_apache', function() {
+    $command = 'service apache2 reload';
+    $command = get('varnish_use_sudo') ? 'sudo '.$command : $command;
```
Comment (RandyCupic) [inline deploy/tasks/server.php]: @iherak You have a typo here :)

**Comment by iherak** [comment]
@emodric @RandyCupic @ludwig031 this PR should now be ok to test and review again.

**Comment by MarioBlazek** [comment]
What is the status of this PR?

**Comment by MarioBlazek** [comment]
@iherak @ludwig031 is this ready for merge?

**Comment by pboethig** [comment]
Edi Modrić <notifications@github.com> schrieb am Mo., 17. Aug. 2020, 11:53:

**Pull request merged**